### ISSUE TITLE
Towards a Docs Dev Framework

### ISSUE THREAD

**Issue by briandominick** [DocOps/wtd2019#5]
Docs-as-code documentation repos are applications, and it's time we recognize it. We should be using standard technology stacks and development frameworks so we can get started faster and achieve more. Topics to cover:

* choosing a tech stack
* boilerplate files and libraries
* file and content architecture
* conventions and namespaces
* open sourcing and sharing

This session would likely work best in the Unconference, as I'm not sure there is an existing expert on the subject. This is the beginning of an important conversation.

**Comment by diggerlilly** [comment]
+1000 on this one

**Comment by diggerlilly** [comment]
Some additional thoughts:
- This relates directly to the ['Wandering the wilderness'](https://github.com/DocOps/wtd2019/issues/7) issue. 
- Newbie thoughts:
-- "I know there are a bunch of people out there doing this but I don't really have someone to show me how..."
-- "I see many examples but I don't understand what requirements went into creating them and I can't find a generic version that I could modify for my own project."
- As a community, we should have a set of frameworks that we can reasonably expect will work for a majority of cases. 
- Can customize from there to handle additional complexity.
- Not a place for competition...should be a variety of options but presented with this same 'framework' mindset

**Comment by mojavelinux** [comment]
To take this up a level, I think it's more about an information architecture / system design than it is a framework. A framework to me implies a solution. But before we have a specific solution, there needs to be an architecture. For example, how are versions managed? A natural fit for docs-as-code is to use references in a VCS. Should it be a tag or a branch? If it's a branch, should it be the same branch that's use for the code or a dedicated one? Should the docs version have the same granularity as the software version, or a broader one? How should files in these branches be aggregated together to build the documentation? How are references made / expressed between the versions? How are changes managed across versions? Where does content go that isn't versioned?

So if we do talk about a framework, I think it should be an architectural framework. Then we can talk about which software frameworks implement / accommodate that architecture. We should really be thinking about the flow of files and information from source to publication.

**Comment by briandominick** [comment]
Yeah I'm not talking about adopting someone else's software framework, so differences in how we see this might just be semantics. I am talking about an application development approach that includes architecture and conventions like you're talking about. It would be set up to "initialize" an environment and have utilities and libraries that enable rapid development and content contributions. It's a concrete set of strategies and resources for implementing complex documentation solutions for different problems, so we don't have to reinvent the wheel each time. I don't much care what we call it, but [the one Alicia and I have been developing](https://github.com/DocOps/liquidoc-cmf) feels like a framework (and has versioning and file architecture built in).
